ipalib/cli.py

```diff
@@ -1454,10 +1454,36 @@ def format_description(self, description):
 )
 
 
+def acquire_cred_interactively(api, principal, ccache_name, service):
+    from gssapi import Name, NameType
+    from gssapi.raw import (acquire_cred_with_password,
+                            store_cred, store_cred_into)
+    if principal is None:
+        principal = api.Backend.textui.prompt('Username')
+    password = api.Backend.textui.prompt_password('Password', False)
```

Comment:
You may want to take `api.env.interactive` and `api.env.prompt_all` into account.

Reply:
`api.env.in_server` would not be needed to check because this callback will not be called in that case. KerbTransport is only used when we have HTTP transport -- in `in_server` we don't use it.

Reply:
I added check for interactivity into `ipalib/rpc.py` and also into the callback. So we wouldn't call into the callback at all in `ipa -n ping` case and will simply fail without creds:
```
ipa: ERROR: did not receive Kerberos credentials
```